producer/src/oid4vc_core.rs

```diff
@@ -70,42 +67,47 @@ impl Verify for SecretManager {
                 DIDUrlQuery::from(&did_url),
                 Some(identity_iota::verification::MethodScope::VerificationMethod),
             )
-            .ok_or(Error::new(
-                ErrorKind::NotFound,
-                format!(
-                    "No verification method found for fragment=[{}]",
-                    did_url.fragment().unwrap()
-                ),
-            ))?;
-
-        let public_key_jwk = verification_method
-            .data()
-            .public_key_jwk()
-            .ok_or(Error::new(ErrorKind::NotFound, "No JWK found"))?;
-
-        let x = match public_key_jwk.params() {
-            identity_iota::verification::jwk::JwkParams::Okp(okp) => okp.x.as_str(),
-            identity_iota::verification::jwk::JwkParams::Ec(ec) => ec.x.as_str(),
-            identity_iota::verification::jwk::JwkParams::Rsa(_) => todo!(),
-            identity_iota::verification::jwk::JwkParams::Oct(_) => todo!(),
-        };
+            .ok_or(ProducerError::Generic(format!(
+                "No verification method found for fragment=[{}]",
+                did_url.fragment().unwrap()
+            )))?;
 
-        Ok(URL_SAFE_NO_PAD.decode(x.as_bytes()).unwrap())
```

Comment:
Now the public_key is not decoded anymore, did_manager will generate invalid material in combination with oid4vc.

Is this a deliberate choice to remove the decoding? Do you think that this decoding should be moved to the oid4vc crates instead?